sourcecode/hub/app/Http/Controllers/UserController.php

```diff
@@ -55,4 +55,26 @@ public function savePreferences(SavePreferencesRequest $request): RedirectRespon
 
         return to_route('user.preferences');
     }
+
+    public function myAccount(): View|RedirectResponse
+    {
+        return view('user.my-account');
+    }
+
+    public function changeUsername(): View|RedirectResponse
+    {
+        return view('user.my-account');
+    }
+
+    public function saveUsername(StoreUserRequest $request): RedirectResponse
+    {
+        $validatedData = $request->validated();
+
+        /** @var User $user */
+        $user = Auth::user();
+
+        $user->update(['name' => $validatedData['name']]);
+
+        return redirect()->route('user.my-account')->with('success', 'Name updated successfully');
```

Comment:
The message must be translatable, and `alert` should be used instead of `success` so the base template can show the message.